src/radical/pilot/utils/component.py

```diff
@@ -1298,17 +1033,15 @@ def advance(self, things, state=None, publish=True, push=False, ts=None,
         otherwise, *only the state* is published.
 
         This is evaluated in self.publish.
-        """
-
-        self.is_valid()
+        '''
 
         if not ts:
             ts = time.time()
 
         if not isinstance(things, list):
             things = [things]
 
-        self._log.debug('advance bulk size: %s [%s, %s]', len(things), push, publish)
+        self._log.debug('advance bulk: %s [%s, %s]', len(things), push, publish)
```

Comment:
Do we need this or is it a left over from debugging?

Reply:
Not sure I understand: it is a debug log entry, so it is here for debugging.  It is not semantically meaningful, so we don't *need* this (apart from debugging).  What am I missing?

Reply:
I noticed that in previous cases you preferred to remove debug statements instead of leaving them around in the code. Maybe this is different? It is logged only when logging is set to DEBUG by the user?

Reply:
Ah, I see.  I sometimes add additional localized debug messages for a specific debugging session and need to clean up afterwards.  Those need to be cleaned out because they don't provide sufficient context to be useful outside of that specific session, add too much noise, or are high-frequency and impact performance (the latter usually I just comment out so they remain somewhat available).  

The one above is a low frequency log which gives reasonable information about task flux for that component, so it can well stay.

Thanks for checking and context! :-)